src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1922,6 +1922,10 @@ bool AdaptDispatch::HardReset()
     // Cursor to 1,1 - the Soft Reset guarantees this is absolute
     success = CursorPosition(1, 1) && success;
 
+    // Reset the mouse mode
+    success = EnableSGRExtendedMouseMode(false);
```

Comment:
Blocker: success is overwritten here.

Reply:
Good catch. For the most part, the earlier methods won't fail... but we should make sure we do this correctly anyway.

Reply:
For the sake of readability, you might want to use the &= operator rather than that foo = bar && foo pattern here. 
Just a suggestion.

Reply:
There's actually a specific reason we're doing it this way. We still want these methods executed, regardless of the state of `success`. So 

```c++
success = foo() && success;
```

will still execute `foo()` even if `success` is already `false`. However,

```c++
success &= foo();
```

will [short-circuit](https://stackoverflow.com/a/5211975/1481137) `foo()` entirely.

Reply:
If I'm not mistaken `&=` is a bitwise operator and will not short-circuit.

Reply:
🤦

Reply:
At any rate, we somewhat strictly observe a rule of _not using bitwise operators on logical values._ Even if it is not specified in ISOCPP, it has been a good rule so far.

Reply:
(A possible explanation that, if asked, I could backfill: (1) the representation of `BOOL` may not lend itself to bitwise operations in the same way as C++'s `bool` and (2) expressions joined with logical operators convert to logical values, leaving `x = x && functionThatReturnsAPointer()` unambiguous with regards to the bitwise representation of the pointer.)